prow/cmd/deck/static/prow/prow.ts

```diff
@@ -674,6 +704,127 @@ function stateToAdj(state: JobState): string {
+    if (startEl != null) {
+        startEl.textContent = `${formatDuration(end - start)} ago`;
+    }
+
+    // make sure the empty table is hidden
+    const tableEl = document.getElementById("job-histogram") as HTMLTableElement;
+    const labelsEl = document.getElementById("job-histogram-labels") as HTMLDivElement;
+    if (jobHistogram.length() === 0) {
+        tableEl.style.display = "none";
+        labelsEl.style.display = "none";
+        return;
+    }
+    tableEl.style.display = "";
+    labelsEl.style.display = "";
+
+    const el = document.getElementById("job-histogram-content") as HTMLTableSectionElement;
+    el.title = `Showing ${jobHistogram.length()} builds from last ${formatDuration(end - start)} by start time and duration, newest to oldest.`;
+    const rows = 10;
+    const width = 12;
+    const cols = Math.round(el.clientWidth / width);
+
+    // initialize the table if the row count changes
+    if (el.childNodes.length !== rows) {
+        el.innerHTML = "";
+        for (let i: number = 0; i < rows; i++) {
+            const tr = document.createElement('tr');
+            for (let j: number = 0; j < cols; j++) {
+                const td = document.createElement('td');
+                tr.appendChild(td);
+            }
+            el.appendChild(tr);
+        }
+    }
+
+    // populate the buckets
+    const buckets = jobHistogram.buckets(start, end, cols);
+    buckets.data.forEach((bucket, colIndex) => {
+        let lastRowIndex = 0;
+        buckets.linearChunks(bucket, rows).forEach((samples, rowIndex) =>  {
+            lastRowIndex = rowIndex + 1;
+            const td = el.childNodes[rows - 1 - rowIndex].childNodes[cols - colIndex - 1] as HTMLTableCellElement;
+            if (samples.length === 0) {
+                td.removeAttribute('title');
+                td.className = '';
+                return;
+            }
+            td.setAttribute('x-data-sample-row', String(samples[0].row));
+            const failures = samples.reduce((sum, sample) => {
+                return sample.state !== 'success' ? sum + 1 : sum;
+            }, 0);
+            if (failures === 0) {
+                td.title = `${samples.length} succeeded`;
+            } else {
+                if (failures === samples.length) {
+                    td.title = `${failures} failed`;
+                } else {
+                    td.title = `${failures}/${samples.length} failed`;
+                }
+            }
+            td.style.opacity = String(0.2 + samples.length / bucket.length * 0.8);
+            if (samples[0].row !== -1) {
+                td.className = `active success-${Math.floor(10 - (failures / samples.length) * 10)}`;
+            } else {
+                td.className = `success-${Math.floor(10 - (failures / samples.length) * 10)}`;
+            }
+        });
+        for (let rowIndex: number = lastRowIndex; rowIndex < rows; rowIndex++) {
```

Comment:
`number` is inferred here and can be omitted.

```suggestion
        for (let rowIndex = lastRowIndex; rowIndex < rows; rowIndex++) {
```